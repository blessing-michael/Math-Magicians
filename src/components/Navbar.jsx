import React from 'react';
import { NavLink } from 'react-router-dom';
import '../styles/Navbar.css';

const links = [
  { path: '/', text: 'Home' },
  { path: 'CalculatorPage', text: 'Calculator' },
  { path: 'Quote', text: 'Quote' },
];

const Navbar = () => (
  <nav className="navbar">
    <h1>Maths Magicians</h1>
    <ul className="nav-menu">
      {links.map((link) => (
        <li key={link.text}>
          <NavLink to={link.path}>{ link.text }</NavLink>
        </li>
      ))}
    </ul>
  </nav>
);

export default Navbar;
