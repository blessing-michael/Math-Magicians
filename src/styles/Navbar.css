* {
  margin: 0;
}

.navbar {
  display: flex;
  align-items: center;
  justify-content: space-between;
  background: grey;
  padding: 1rem 2rem;
  margin: 0 auto;
  box-shadow: 0 4px 14px 0 rgba(70, 70, 70, 0.38);
}

.nav-menu {
  font-size: 1.4rem;
  display: flex;
  list-style: none;
  gap: 2rem;
}
