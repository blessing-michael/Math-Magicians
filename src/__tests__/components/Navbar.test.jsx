import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Navbar from '../../components/Navbar';

describe('Navbar', () => {
  test('renders the Navbar component', () => {
    const { container } = render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
