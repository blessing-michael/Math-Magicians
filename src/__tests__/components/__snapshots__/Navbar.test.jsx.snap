// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders the Navbar component 1`] = `
<nav
  class="navbar"
>
  <h1>
    Maths Magicians
  </h1>
  <ul
    class="nav-menu"
  >
    <li>
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        Home
      </a>
    </li>
    <li>
      <a
        class=""
        href="/CalculatorPage"
      >
        Calculator
      </a>
    </li>
    <li>
      <a
        class=""
        href="/Quote"
      >
        Quote
      </a>
    </li>
  </ul>
</nav>
`;
